linfa-svm/src/regression.rs

```diff
@@ -223,7 +227,7 @@ impl<D: Data<Elem = f64>> Predict<ArrayBase<D, Ix2>, Vec<f64>> for Svm<f64, f64>
             .map(|data| {
                 let val = match self.linear_decision {
                     Some(ref x) => x.mul(&data).sum() - self.rho,
-                    None => self.kernel.weighted_sum(&self.alpha, data.view()) - self.rho,
+                    None => self.weighted_sum(data.view()) - self.rho,
```

Comment:
I think we could use an enum here like the one used for the kernel inner matrix so that it either holds the precomputed weights in the linear case or either the support vectors. In that way, we could avoid storing the support vectors in the linear case. I can't think of what could be an appropriate name for such a structure tough

Reply:
Yes, that sounds like a good name for it